src/app/app.component.html

```diff
@@ -0,0 +1,18 @@
+<mat-toolbar color="primary">
+  <a class="title" href="/">{{ title }}</a>
+
+  <div class="toolbar-spacer"></div>
+  
+  <button mat-raised-button *ngIf="!loggedIn" (click)="login()">Login</button>
+  <button mat-raised-button color="warn" *ngIf="loggedIn" (click)="logout()">Logout</button>
+</mat-toolbar>
+
+<mat-card class='card-section'>
+  <mat-card-title>Sign-in with Microsoft Azure AD B2C</mat-card-title>
+  <mat-card-subtitle>Getting an access token with Azure AD B2C and calling a Web API</mat-card-subtitle>
+  <mat-card-content>This simple sample demonstrates how to use the Microsoft Authentication Library for JavaScript (msal.js) to get an access token and call an API secured by Azure AD B2C.</mat-card-content>
+  <mat-card-actions>
+    <button mat-raised-button *ngIf="loggedIn && !tokenAcquired" (click)="getProfile()">Call API</button>
+  </mat-card-actions>
+  <mat-card-content id="apiResponse" *ngIf="tokenAcquired"> Welcome {{apiResponse?.name}}! </mat-card-content>
+</mat-card>
```

Comment:
hmm, this caused the ui to load twice -perhaps it was meant to be *ngIf="isIframe"?

![Capture](https://user-images.githubusercontent.com/13197548/75194087-507fd380-570c-11ea-8e17-91692b01e27d.PNG)

Reply:
What does your code look like?

Reply:
I followed the link above verbatim, though I can't say I quite understand the logic here

I have these (just pushed the changes):

```javascript

const routes: Routes = [
  {
    path: '',
    component: AppComponent
  }
];

this.isIframe = window !== window.parent && !window.opener;
```
```html
<div class="container">
  <!--This is to avoid reload during acquireTokenSilent() because of hidden iframe -->
  <router-outlet *ngIf="!isIframe"></router-outlet>
</div>
```

Reply:
Ah, instead of the `AppComponent` for the `component`, you should create a separate `HomeComponent` with the content of the home screen (the `mat-card-` components) and then remove this code from `AppComponent`. 

Example:
https://github.com/Azure-Samples/active-directory-javascript-singlepageapp-angular/blob/master/src/app/app-routing.module.ts
https://github.com/Azure-Samples/active-directory-javascript-singlepageapp-angular/blob/master/src/app/app.component.html

Reply:
Oh, makes sense -will do